modules/cachedb_redis/cachedb_redis_utils.c

```diff
@@ -114,12 +114,14 @@ cluster_node *get_redis_connection_by_endpoint(redis_con *con, redis_moved *redi
 		return con->nodes;
 	} else {
 		for (it=con->nodes;it;it=it->next) {
-            if (match_prefix(redis_info->endpoint.s, redis_info->endpoint.len, it->ip, strlen(it->ip))) {
+			if (match_prefix(redis_info->endpoint.s, redis_info->endpoint.len, it->ip, strlen(it->ip))) {
 				if (it->port == redis_info->port) {
-					if (it->start_slot <= redis_info->slot && it->end_slot >= redis_info->slot) {
+					// Removed slot comparison as it may be a little too aggressive of a match
+					// Code is still here in the event that it needs to be added back in
+					//if (it->start_slot <= redis_info->slot && it->end_slot >= redis_info->slot) {
```

Comment:
Agree on removing the slot comparison for now, until the cluster "topology scanning" support is extended to be also performed at runtime, perhaps a one-time operation after each MOVED reply, ideally.

But isn't there still an issue with your MOVED implementation whenever we add a +1 node to the Redis Cluster?  All of a sudden, some of the OpenSIPS queries get a MOVED redirection to the new cluster node (the keys are balanced across all nodes).  But this new function you added, `get_redis_connection_by_endpoint()` tries to match the new endpoint against the existing list of nodes, which is the incomplete one (before the +1 node was added).  Won't the lookup fail?  Please correct me if I'm wrong.